activesupport/lib/active_support/core_ext/hash/keys.rb

```diff
@@ -134,6 +140,12 @@ def _deep_transform_keys_in_object!(object, &block)
           object[yield(key)] = _deep_transform_keys_in_object!(value, &block)
         end
         object
+      when defined?(ActionController::Parameters) && ActionController::Parameters
```

Comment:
Active Support should not have any knowledge about `ActionController`. Can't this problem be solve by overriding those methods in `ActionController::Parameters`?